### PR TITLE
Shadcn migration - DeveloperDocsLinks

### PR DESCRIPTION
**Pull request by saurabhburade**
## Description
Migrate `DeveloperDocsLinks` to tailwind.

## Related Issue
#13946

### CODE REVIEW

File: src/components/DeveloperDocsLinks.tsx [hunk 1]
```diff
@@ -15,10 +15,10 @@ const DeveloperDocsLinks = ({ headerId }: DeveloperDocsLinksProps) => (
     {docLinks
       .filter(({ id }) => id.includes(headerId))
       .map(({ items, id }) => (
-        <UnorderedList ms={6} spacing={3} key={id}>
+        <ul className="ms-6 space-y-3" key={id}>
```
Comment (pettinarip) [inline src/components/DeveloperDocsLinks.tsx:18]: Lets replace this with the `UnorderedList` and `ListItem` defined in [this file](https://github.com/ethereum/ethereum-org-website/blob/dev/src/components/ui/list.tsx)

**Review by pettinarip — CHANGES_REQUESTED** [review]
Looking good, small request to use our ui components

File: src/components/DeveloperDocsLinks.tsx (hunk 1 above)
Comment (saurabhburade) [reply src/components/DeveloperDocsLinks.tsx:18]: Got it.

**Review by pettinarip — APPROVED** [review]
lgtm!

**Pull request merged**